test/testautomation_platform.c

```diff
@@ -386,7 +386,7 @@ int platform_testSetErrorEmptyInput(void *arg)
 int platform_testSetErrorInvalidInput(void *arg)
 {
     int result;
-    const char *invalidError = NULL;
+    const char *invalidError = "";
```

Comment:
Neither SDL3 nor SDL3 do this: why is this needed?

Reply:
If this is really needed, it should be done in SDL2 and SDL3 first, and done here as a separate patch

Reply:
```
test/testautomation_platform.c: In function ‘platform_testSetErrorEmptyInput’:
test/testautomation_platform.c:351:28: error: ‘%s’ directive argument is null [-Werror=format-overflow=]
  351 |     result = SDL_SetError("%s", testError);
      |                            ^~
test/testautomation_platform.c:364:64: error: ‘%s’ directive argument is null [-Werror=format-overflow=]
  364 |                             "SDL_GetError(): expected message '%s', was message: '%s'",
```

was just fixing compilation also for me
 (this should not be part of this PR)

Reply:
revert, done in other pr